ui/app/components/ui/tabs/tabs.component.js

```diff
@@ -29,11 +37,11 @@ export default class Tabs extends Component {
     const numberOfTabs = React.Children.count(this.props.children)
 
     return React.Children.map(this.props.children, (child, index) => {
+      const tabKey = this._generateKey(child)
       return child && React.cloneElement(child, {
-        onClick: (index) => this.handleTabClick(index),
+        onClick: (index) => this.handleTabClick(index, tabKey),
         tabIndex: index,
         isActive: numberOfTabs > 1 && index === this.state.activeTabIndex,
-        key: index,
```

Comment:
I believe this isn't respected, per the docs:<sup>[\[1\]][1]</sup>

> Clone and return a new React element using `element` as the starting point. The resulting element will have the original element’s props with the new props merged in shallowly. New children will replace existing children. `key` and `ref` from the original element will be preserved.

Also, key isn't a prop, it's a reconciling hint, and this 2nd argument is for props.<sup>[\[2\]][2]</sup>

  [1]:https://reactjs.org/docs/react-api.html#cloneelement
  [2]:https://reactjs.org/warnings/special-props.html